### ISSUE TITLE
[DatePicker]: aria-label format issue for dates

### ISSUE THREAD

**Issue by shubh1502** [UI5/webcomponents#13594]
### Bug Description

Description:
We identified an accessibility inconsistency in the DatePicker calendar between SAPUI5 framework-based controls and UI5 Web Components.

In SAPUI5 framework-based DatePicker controls, the calendar day cells expose the aria-label in the following format:

Day Month Year
Example: 12 March 2026

However, in UI5 Web Components DatePicker, the calendar day cells expose the aria-label in a different format:

Month Day Year
Example: March 12 2026

This inconsistency was reported by our accessibility team because users of assistive technologies experience different reading patterns depending on which DatePicker implementation is used.

After investigating the issue, we found that the formatting appears to originate from the UI5 Web Components implementation itself.

### Affected Component

DatePicker/Calender

### Expected Behaviour

Expected Behavior:
The aria-label format in UI5 Web Components DatePicker should be aligned with the SAPUI5 framework implementation for consistency and accessibility compliance.

Expected format:

Day Month Year

Actual Behavior:
UI5 Web Components DatePicker uses:

Month Day Year

### Isolated Example

https://ui5.github.io/webcomponents/components/DatePicker/

### Steps to Reproduce

1.
2.
3.
...

### Priority

Medium

### UI5 Web Components Version

2.22.0

### Browser

Chrome

### Organization

SAP Labs

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by nikoletavnv** [comment]
Hi @shubh1502, 
Can you give me an example where the aria-label format for the calendar date picker in SAPUI5 controls is in the format "Day Month Year"? In the UI5 demo kit (https://ui5.sap.com/#/entity/sap.m.DatePicker/sample/sap.m.sample.DatePicker) the dates from the picker have the same format "Month Day, Year" as in the Web component. 
Thanks!
Kind regards, 
Nikoleta

**Comment by shubh1502** [comment]
Hi @nikoletavnv ,

I am attaching a screenshot where aria-label can be seen in format "day month, year" from the same UI5 demo kit link attached above.

<img width="1919" height="1058" alt="Image" src="https://github.com/user-attachments/assets/cb8a6c49-6bb0-4854-a910-ee9491605509" />,

whereas the aria-label for web component is in format "Month day, year"

<img width="1917" height="1132" alt="Image" src="https://github.com/user-attachments/assets/496daf89-2ce5-43b9-b56f-2e6da2d5d330" />

Thanks,
Shubham

**Comment by nikoletavnv** [comment]
Dear @UI5/ui5-team-ctr-bal , 

In the web components the aria-label for a day in the date picker differs in format with the aria-label in the UI5 controls. The format in the web components is fixed and is not dependent on the localization.

Kind regards, 
Nikoleta
